### PR TITLE
fix(dropdown): default selected not getting selected & removing options prop

### PR DESCRIPTION
**Pull request by mJarsater**
**Describe pull-request**  
The default-selected prop wasn't working properly, this due to us not filtering the dropdowns children to only be tds-dropdown-option elements.

BREAKING
Removed the options property.

**Solving issue**
Fixes: [CDEP-2439](https://tegel.atlassian.net/browse/CDEP-2439)

**How to test**  
1. Go to storybook
2. Check in Dropdown
3. Try the default selected controls for all dropdown versions.

**Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

[CDEP-2439]: https://tegel.atlassian.net/browse/CDEP-2439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by TatyanaBumazhkina** [comment]
Tested and passed in Chrome, Edge, Firefox, Safari browsers.
Both cases described in the bug ticket work properly.

File: core/src/components/dropdown/dropdown-option/dropdown-option.tsx [hunk 1]
```diff
@@ -31,12 +31,10 @@ export class TdsDropdownOption {
 
   private label: string;
 
-  /** Method to select/deselect an option if the option is not disabled. */
+  /** Method to select/deselect an option if the option. */
```
Comment (nathalielindqvist) [inline core/src/components/dropdown/dropdown-option/dropdown-option.tsx:34]:
```suggestion
  /** Method to select/deselect an option. */
```

File: core/src/components/dropdown/dropdown.stories.tsx [hunk 2]
```diff
@@ -177,6 +177,7 @@ const defaultOptionLookUp = {
   'Option 1': 'option-1',
   'Option 2': 'option-2',
   'Option 3': 'option-3',
+  'Option 4': 'option-4',
```
Comment (nathalielindqvist) [inline core/src/components/dropdown/dropdown.stories.tsx:180]: When multiselect is active and option types are rendered via the options prop there is an option to put "Option 4" as a default option, but it doesn't exist in the Dropdown itself. Might be an edge case though so not sure it's worth addressing at this point?
image

**Review by nathalielindqvist — DISMISSED** [review]
Noice! ✨

**Review by nathalielindqvist — APPROVED** [review]

**Review by theJohnnyMe — APPROVED** [review]
LGTM!

**Pull request merged**